packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputMultipleSingleSelect.tsx

```diff
@@ -27,11 +27,12 @@ function GridFilterInputMultipleSingleSelect(props: GridFilterInputMultipleSingl
     }
   }
 
-  const getOptionValue = resolvedColumn?.getOptionValue!;
-  const getOptionLabel = resolvedColumn?.getOptionLabel!;
+  const getOptionValue = resolvedColumn?.getOptionValue;
+  const getOptionLabel = resolvedColumn?.getOptionLabel;
 
   const isOptionEqualToValue = React.useCallback(
-    (option: ValueOptions, value: ValueOptions) => getOptionValue(option) === getOptionValue(value),
+    (option: ValueOptions, value: ValueOptions) =>
+      getOptionValue?.(option) === getOptionValue?.(value),
```

Comment:
tbh, this still feels wrong, this would always return `true` if `getOptionValue` is undefined, ifen if `value` and `option` are not equivalent. If there's a valid reason where `resolvedColumn?.getOptionLabel` can be `undefined`, shouldn't it be more like:

```tsx
const getOptionValue = resolvedColumn?.getOptionValue ?? (value) => value;
const getOptionLabel = resolvedColumn?.getOptionLabel ?? (label) => label;
```

?

Otherwise can we express it correctly in the types?

Reply:
OK, should be done.
Basically, if `resolvedColumn` is not a single select column, we should just return null.
I've made `getOptionLabel` and `getOptionValue` non-nullable, because we assign default ones anyway.